sdk/core/azure-core/src/main/java/com/azure/core/models/CloudEvent.java

```diff
@@ -228,9 +280,21 @@ private CloudEvent() {
             }
             for (CloudEvent event : events) {
                 if (event.getId() == null || event.getSource() == null || event.getType() == null) {
+                    List<String> nullAttributes = new ArrayList<String>();
+                    if (event.getId() == null) {
+                        nullAttributes.add("'id'");
+                    }
+                    if (event.getId() == null) {
+                        nullAttributes.add("'source'");
+                    }
+                    if (event.getId() == null) {
+                        nullAttributes.add("'type'");
+                    }
```

Comment:
Long term, for performance reasons, it may be cheaper to have a large `if/else if` block with the error messages constructed at startup.

Reply:
Let's follow up in #19679